src/main/java/emissary/core/channels/FileChannelFactory.java

```diff
@@ -49,47 +43,9 @@ private FileChannelFactoryImpl(final Path path) {
          * @return the new channel instance
          */
         @Override
-        public SeekableByteChannel create() {
-            return new LazyFileChannelImpl(path);
+        public LazyOpeningFileChannel create() {
+            return new LazyOpeningFileChannel(path);
         }
     }
 
-    private static final class LazyFileChannelImpl extends AbstractSeekableByteChannel {
```

Comment:
Another example when I pulled back the curtain some and broke some of the encapsulation.